pkg/tsdb/prometheus/querydata/response_test.go

```diff
@@ -0,0 +1,58 @@
+package querydata
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/grafana/grafana/pkg/tsdb/prometheus/models"
+	"github.com/grafana/grafana/pkg/tsdb/prometheus/querydata/exemplar"
+)
+
+func TestQueryData_parseResponse(t *testing.T) {
+	qd := QueryData{exemplarSampler: exemplar.NewStandardDeviationSampler}
+
+	t.Run("resultType is before result the field must parsed normally", func(t *testing.T) {
+		resBody := "{\"data\":{\"resultType\":\"vector\", \"result\":[{\"metric\":{\"__name__\":\"some_name\",\"environment\":\"some_env\",\"id\":\"some_id\",\"instance\":\"some_instance:1234\",\"job\":\"some_job\",\"name\":\"another_name\",\"region\":\"some_region\"},\"value\":[1.1,\"2\"]}]},\"status\":\"success\"}"
```

Comment:
Fore these, you can replace the outer double quotes `"` with a backtick (backquote) , and then all the escaping isn't needed.

Reply:
I did that and all tests failed. Let me try that once more. Maybe I did something wrong.

Reply:
I made a mistake earlier. Now they are with backticks.